### ISSUE TITLE
Reloaded setInterval fails

### ISSUE THREAD

**Issue by omphalos** [omphalos/thermite#4]
``` js
test('should hot swap nested setInterval', function(t) {
  var state = { counter: 0 }
  var target = thermite.eval('(' + function outer(state) {
    return 0
  } + ')')

  target.hotSwap('(' + function outer(state) {
    state.si = state.si || setInterval(function inner() {
      state.counter++
    }, 1)
  } + ')')
  target.result(state)

  setTimeout(function() {
    target.hotSwap('(' + function outer(state) {
      state.si = state.si || setInterval(function inner() {
        state.counter = 'x'
      }, 1)
    } + ')')
    target.result(state)

    setTimeout(function() {
      t.equal(state.counter, 'x')
      clearInterval(state.si)
      t.end()
    }, 100)
  }, 100)
})
```

**Comment by omphalos** [comment]
Simplified test:

``` js
it('should hot swap nested function twice', function() {
  var target = thermite.eval('(' + function outer() {
    return function inner() {}
  } + ')')

  target.hotSwap('(' + function outer() {
    return function inner() { return 'a' }
  } + ')')
  var callback = target.result()

  target.hotSwap('(' + function outer() {
    return function inner() { return 'b' }
  } + ')')

  assert.equal(callback(), 'b')
})
```

**Comment by omphalos** [comment]
Fixed in 0.0.6.
